### PR TITLE
Dataset sap

### PR DESCRIPTION
**Pull request by Bonnarel**
Major change to change the scope from images/cubes to all datasets. Solves issue #10

### CODE REVIEW

File: SIA.tex [hunk 1]
```diff
@@ -52,45 +45,22 @@ \section{Introduction}
...
-    \item  The parameter language should be extended to support case-insensitivity, wildcards, and pattern recognition.
-    \item The DALI UPLOAD facility will be defined in order to allow queries based on values stored in a file.
-    \item The protocol will also support an option for virtual data discovery. With this option, controlled by usage of an optional parameter, the  query can force the discovery of virtual datasets with best  matching to the input parameter constraints
-    \item COLLECTION and FACILITY currently provide query parameters provide selection on service defined set of strings. It would be good to define standardized lists of those at IVOA level for next version.
-    \item Visibility data and event lists are not considered as valid DPTYPES in this version due to complexity of data access functionalities for those dataset types.  
-\end{itemize}
+DsSAP specifies standardID values for each capability, as defined by VODataService \citep{std:VODS11}. DsSAP services may be registered in an IVOA Registry using the SimpleDALRegExt \citep{std:DALREGEXT} extension schema.
+\subsection{Changes from SIA-2.0i to DsSAP}
```
Comment (jd-au) [inline SIA.tex]: Is the i on SIA-2.0i intended?

File: SIA.tex [hunk 2]
```diff
@@ -52,45 +45,22 @@ \section{Introduction}
...
 
+Virtual Observatory access to astronomical images has been available via the SIA-1.0 protocol for over a decade.  Many such services have been implemented since 2002, and SIA-1.0 \citep{std:SIAP} was formally standardized as an IVOA Recommendation in 2009. SSA \citep{std:SSA} played a similar role for spectra and also used specific metadata in the query response. SIA-2.0 \citep{std:SIA2} was multi-dimensional and fully integrated with the modern VO architecture and related standards, but restricted to images and data cubes.
+DsSAP is an extension of SIA2 to other dataproduct types. It can be seen as a server side  parameter based proxy to an ObsTAP service.
+IA-2.0 differs from legacy SIA-1.0 in the following aspects:
```
Comment (jd-au) [inline SIA.tex]: Is there a list of differences missing here?

File: SIA.tex [hunk 3]
```diff
@@ -52,45 +45,22 @@ \section{Introduction}
...
-    \item COLLECTION and FACILITY currently provide query parameters provide selection on service defined set of strings. It would be good to define standardized lists of those at IVOA level for next version.
-    \item Visibility data and event lists are not considered as valid DPTYPES in this version due to complexity of data access functionalities for those dataset types.  
-\end{itemize}
+DsSAP specifies standardID values for each capability, as defined by VODataService \citep{std:VODS11}. DsSAP services may be registered in an IVOA Registry using the SimpleDALRegExt \citep{std:DALREGEXT} extension schema.
+\subsection{Changes from SIA-2.0i to DsSAP}
 
+Virtual Observatory access to astronomical images has been available via the SIA-1.0 protocol for over a decade.  Many such services have been implemented since 2002, and SIA-1.0 \citep{std:SIAP} was formally standardized as an IVOA Recommendation in 2009. SSA \citep{std:SSA} played a similar role for spectra and also used specific metadata in the query response. SIA-2.0 \citep{std:SIA2} was multi-dimensional and fully integrated with the modern VO architecture and related standards, but restricted to images and data cubes.
```
Comment (jd-au) [inline SIA.tex]: long lines like this should be broken up to assist readability of the code. This applies to quite a bit of the new content.

Comment (jd-au) [inline SIA.tex]: The `std:SSA` and `std:SIA2` citations are unresolved and come up as `?`

File: SIA.tex [hunk 4]
```diff
@@ -104,26 +74,27 @@ \subsubsection{Simple Data Discovery}
...
+In many cases, datasets are too large to download and process locally, so the client must be able to perform remote operations. Data discovery could be performed using any discovery protocol (DsSAP, SIA, TAP with ObsCore, etc.).  The client must be able to easily figure out if a low level access service is available for a discovered dataset. This could be using a URL provided in the response or by calling an associated DataLink service. Access operations include basic filtering (cut out a subsection of the data), transformations, or other pixel-level operations or even analysis. With version 1.0 of the SODA 1.0  specification, we will only cover extracting a simple subset of an image or datacube.
```
Comment (jd-au) [inline SIA.tex]: The 1.0 in SODA 1.0 is repeated and should this be replaced with a citation of SODA 1.0?

File: ivoatexmeta.tex [hunk 5]
```diff
@@ -0,0 +1,6 @@
+% GENERATED FILE -- edit this in the Makefile
```
Comment (jd-au) [inline ivoatexmeta.tex]: Should this file be included in the PR?

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+name: Update PDF Preview
```
Comment (jd-au) [inline .github/workflows/main.yml]: How does this relate to the existing `preview.yml` build script?

File: SIA.tex [hunk 7]
```diff
@@ -472,7 +443,9 @@ \subsubsection{INSTRUMENT}
 The INSTRUMENT parameter is a string-valued parameter that specifies the name of the instrument with which the data was acquired. The value is compared with the instrument\_name from the ObsCore data model.
 
 \subsubsection{DPTYPE}
-The DPTYPE parameter is a string-valued parameter that specifies the type of data. The value is compared with the dataproduct\_type from the ObsCore data model. For the SIA \{query\} resource, the only values that should be returned for dataproduct\_type are \textit{image} and \textit{cube}, so this parameter can be only really be used to select one of these.
+The DPTYPE parameter is a string-valued parameter that specifies the type of data. The value is compared with the dataproduct\_type from the ObsCore data model.  In contrast to SIA2.0 which allows only dataproduct\_types  \textit{image} and \textit{cube}, all the other dataproduct\_types are allowed, in such a way that we can constrain the service to retrieve visibility data, timeseries and event lists. Using DPTYPE with the "spectrum" value can be considered as an upgrade of the SSA protocol.
```
Comment (jd-au) [inline SIA.tex:456]: Should we include catalogs here? CASDA has downloadable catalogs included in obscore but the  dataproduct_type for those must be null according to obscore, which hinders discoverability.

**Review by jd-au — COMMENTED** [review]
Great to see the expansion of SIA to further data types.

**Comment by Bonnarel** [comment]
Le 23/04/2022 à 02:39, James Dempsey a écrit :
>
> ***@***.**** commented on this pull request.
>
> Great to see the expansion of SIA to further data types.
>
Thanks for reviewing this. I'll answer and fix the bugs ASAP !

File: SIA.tex (hunk 1 above)
Comment (Bonnarel) [reply SIA.tex]: no

File: SIA.tex (hunk 2 above)
Comment (Bonnarel) [reply SIA.tex]: No. Have to remove line 59

File: SIA.tex (hunk 3 above)
Comment (Bonnarel) [reply SIA.tex]: OK

File: SIA.tex (hunk 4 above)
Comment (Bonnarel) [reply SIA.tex]: OK. Remove second 1.0

File: ivoatexmeta.tex (hunk 5 above)
Comment (Bonnarel) [reply ivoatexmeta.tex]: No

File: SIA.tex (hunk 7 above)
Comment (Bonnarel) [reply SIA.tex:456]: Tricky point. Please open a new issue for that

**Comment by Bonnarel** [comment]
Changes will no more be provided by bunches of little PRs

**Pull request closed without merging**